### ISSUE TITLE
UnboundLocalError: local variable 'subject_map' referenced before assignment

### ISSUE THREAD

**Issue by nleguillarme** [SDM-TIB/SDM-RDFizer#73]
**Describe the bug**
Running rdfizer on a RML file produced with YARRRML Parser raises UnboundLocalError

INFO:rdflib:RDFLib Version: 4.2.2
Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/leguilln/workspace/KNOWLEDGE_INTEGRATION/inteGraph/rdfizer/rdfizer-env/lib/python3.8/site-packages/rdfizer/__main__.py", line 44, in <module>
    semantify(config_path)
  File "/home/leguilln/workspace/KNOWLEDGE_INTEGRATION/inteGraph/rdfizer/rdfizer-env/lib/python3.8/site-packages/rdfizer/__init__.py", line 4047, in semantify
    triples_map_list = mapping_parser(config[dataset_i]["mapping"])
  File "/home/leguilln/workspace/KNOWLEDGE_INTEGRATION/inteGraph/rdfizer/rdfizer-env/lib/python3.8/site-packages/rdfizer/__init__.py", line 593, in mapping_parser
    current_triples_map = tm.TriplesMap(str(result_triples_map.triples_map_id), str(result_triples_map.data_source), subject_map, predicate_object_maps_list, ref_form=str(result_triples_map.ref_form), iterator=str(result_triples_map.iterator), tablename=str(result_triples_map.tablename), query=str(result_triples_map.query))
UnboundLocalError: local variable 'subject_map' referenced before assignment

**To Reproduce**
Steps to reproduce the behavior (and resources):
1. python3 -m rdfizer -c config.ini

**Expected behavior**
Expected the creation of a N-triple file

**Additional context**
Attached are RML mappings and TSV files
[config.ini.txt](https://github.com/SDM-TIB/SDM-RDFizer/files/7619162/config.ini.txt)
[mapping.rml.txt](https://github.com/SDM-TIB/SDM-RDFizer/files/7619163/mapping.rml.txt)
[taxon.tsv.txt](https://github.com/SDM-TIB/SDM-RDFizer/files/7619164/taxon.tsv.txt)
[s.tsv.txt](https://github.com/SDM-TIB/SDM-RDFizer/files/7619165/s.tsv.txt)

**Comment by nleguillarme** [comment]
Hi, thank you for your help.
I've just tested by adding templates to my subject maps and it works.

Now, I would like to move from a yarrrml-parser + rmlmapper pipeline to yarrrml-parser + rdfizer and I  have two problems :
- I need to transform tsv files to RDF, but it seems that rdfizer only handles comma-separated values. Would it be possible to handle other separators ?
- I want my subjects to be real blank nodes, without URIs, that's why I don't specify a template for my subject maps. In this situation, RMLMapper automatically generates blank nodes (e.g. _:59SyVseNjX). Is it possible using RDFizer, and if not, would it be a possible enhancement ?

**Comment by nleguillarme** [comment]
Hi, do you plan to support blank nodes and TSV files in a future release ?

**Comment by eiglesias34** [comment]
Hello again,

First of all, thank you for using the SDM-RDFizer.

Regarding the issue of the Blank Nodes, we follow the semantic defined by R2RML (which is also true for RML):

_Triples map must have exactly one subject map that specifies how to generate a subject for each row of the logical table. It may be specified in two ways:
using the rr:subjectMap property, whose value must be the subject map, or
using the constant shortcut property rr:subject.t must have exactly one subject map that specifies how to generate a subject for each row of the logical table. It may be specified in two ways:
using the rr:subjectMap property, whose value must be the subject map, or
using the constant shortcut property rr:subject._

https://www.w3.org/TR/r2rml/#dfn-subject-map

In other words, Blank nodes will only be generated if there is subject map with the rr:termType rr:BlankNode. Please note that  it is not necessary for the value in subject map to be a URL in this case. Follow the example in my previous message.

Regarding the TSV files, I'll update the SDM-RDFizer so that it can transform TSV files some point this week.

Sincerely,
Enrique Iglesias

**Comment by eiglesias34** [comment]
I updated the SDM-RDFizer so that it can semantify TSV files.

**Comment by eiglesias34** [comment]
Hello,

Since the problem has been solved I am going to close this issue.

Thank you again for using the SDM-RDFizer.
